packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -101,6 +103,25 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{ sync: DashboardCursor
 
   let indexByName: Map<string, number> | undefined;
 
+  // populate any composite renderers which can be used (where all required semanticType fields exist)
+  const compositeRenderers = new Set<Function>();
+
+  // TODO: actually make this robust
+  for (let i = 1; i < frame.fields.length; i++) {
+    let semType = frame.fields[i].state!.semanticType;
+
+    if (semType) {
+      compositeRenderers.add(renderOHLCV);
+    }
+  }
+
+  if (compositeRenderers.size > 0) {
+    // drawAxes fires after the grid is drawn but before any other series are rendered
+    builder.addHook('drawAxes', (u) => {
+      compositeRenderers.forEach((r) => r(u, frame));
+    });
```

Comment:
Question - can't we render composite after everything else is in place?

Reply:
Also, according to my other comment about semanticType on field state - this code could be driven by the mappings config rather than fields state. I think you could avoid the fields loop if only the config is in place.